tensorflow/java/src/test/java/org/tensorflow/TensorTest.java

```diff
@@ -256,82 +302,114 @@ public void writeTo() {
-          // expected
-        }
-      }
     }
   }
 
   @Test
-  public void scalars() {
-    try (Tensor t = Tensor.create(2.718f)) {
-      assertEquals(DataType.FLOAT, t.dataType());
-      assertEquals(0, t.numDimensions());
-      assertEquals(0, t.shape().length);
-      assertEquals(2.718f, t.floatValue(), EPSILON_F);
+  public void whenWriteToIncompatibleDoubleBufferThenThrowExpectedException() {
+    expectedException.expect(IllegalArgumentException.class);
+    expectedException.expectMessage("cannot use java.nio.HeapDoubleBuffer with Tensor of type BOOL");
+    final boolean[] bools = {true, false, true};
+
+    try (Tensor tensor = Tensor.create(bools)) {
+      final DoubleBuffer badBuffer = DoubleBuffer.allocate(128);
+      tensor.writeTo(badBuffer);
     }
+  }
 
-    try (Tensor t = Tensor.create(3.1415)) {
-      assertEquals(DataType.DOUBLE, t.dataType());
-      assertEquals(0, t.numDimensions());
-      assertEquals(0, t.shape().length);
-      assertEquals(3.1415, t.doubleValue(), EPSILON);
+  @Test
+  public void whenWriteToIncompatibleFloatBufferThenThrowExpectedException() {
+    expectedException.expect(IllegalArgumentException.class);
+    expectedException.expectMessage("cannot use java.nio.HeapFloatBuffer with Tensor of type BOOL");
+    final boolean[] bools = {true, false, true};
+
+    try (Tensor tensor = Tensor.create(bools)) {
+      final FloatBuffer badBuffer = FloatBuffer.allocate(128);
+      tensor.writeTo(badBuffer);
     }
+  }
 
-    try (Tensor t = Tensor.create(-33)) {
-      assertEquals(DataType.INT32, t.dataType());
-      assertEquals(0, t.numDimensions());
-      assertEquals(0, t.shape().length);
-      assertEquals(-33, t.intValue());
+  @Test
+  public void whenWriteToIncompatibleIntBufferThenThrowExpectedException() {
+    expectedException.expect(IllegalArgumentException.class);
+    expectedException.expectMessage("cannot use java.nio.HeapIntBuffer with Tensor of type BOOL");
+    final boolean[] bools = {true, false, true};
+
+    try (Tensor tensor = Tensor.create(bools)) {
+      final IntBuffer badBuffer = IntBuffer.allocate(128);
+      tensor.writeTo(badBuffer);
+    } 
+  }
+
+  @Test
+  public void whenWriteToIncompatibleLongBufferThenThrowExpectedException() {
+    expectedException.expect(IllegalArgumentException.class);
+    expectedException.expectMessage("cannot use java.nio.HeapLongBuffer with Tensor of type BOOL");
+    final boolean[] bools = {true, false, true};
+
+    try (Tensor tensor = Tensor.create(bools)) {
+      final LongBuffer badBuffer = LongBuffer.allocate(128);
+      tensor.writeTo(badBuffer);
     }
+  }
 
-    try (Tensor t = Tensor.create(8589934592L)) {
-      assertEquals(DataType.INT64, t.dataType());
-      assertEquals(0, t.numDimensions());
-      assertEquals(0, t.shape().length);
-      assertEquals(8589934592L, t.longValue());
+  @Test
+  public void whenCreateUsingDoubleThenReturnDoubleScalarWithExpectedValue() {
```

Comment:
Can we shorten these and other test names where possible. For example, this can be just
```java
@Test
public void createScalarDoubleTensor()
```